applications/sdp/mspi/src/hrt/hrt.c

```diff
@@ -6,183 +6,102 @@
-
-	out = nrf_vpr_csr_vio_out_get();
-
-	nrf_vpr_csr_vio_out_set(out | PIN_OUT_LOW_MASK(VIO(NRFE_MSPI_DQ0_PIN_NUMBER)) |
-				PIN_OUT_LOW_MASK(VIO(NRFE_MSPI_DQ1_PIN_NUMBER)) |
-				PIN_OUT_LOW_MASK(VIO(NRFE_MSPI_DQ2_PIN_NUMBER)) |
-				PIN_OUT_LOW_MASK(VIO(NRFE_MSPI_DQ3_PIN_NUMBER)));
-
-	nrf_vpr_csr_vio_mode_out_set(&out_mode);
-	nrf_vpr_csr_vio_mode_in_buffered_set(NRF_VPR_CSR_VIO_MODE_IN_CONTINUOUS);
-
-	nrf_vpr_csr_vio_config_get(&config);
-	config.input_sel = false;
-	nrf_vpr_csr_vio_config_set(&config);
+		out = nrf_vpr_csr_vio_out_get();
 
-	/* Fix position of data if word size < MAX_WORD_SIZE,
-	 * so that leading zeros would not be printed instead of data.
-	 */
-	if (xfer_ll_params.word_size < MAX_WORD_SIZE) {
-		for (uint8_t i = 0; i < xfer_ll_params.data_len; i++) {
-			xfer_ll_params.data_to_send[i] =
-				xfer_ll_params.data_to_send[i]
-				<< (MAX_WORD_SIZE - xfer_ll_params.word_size);
+		if (xfer_ll_params.ce_polarity == MSPI_CE_ACTIVE_LOW) {
+			out = BIT_SET_VALUE(out, xfer_ll_params.ce_vio, VPRCSR_NORDIC_OUT_HIGH);
+		} else {
+			out = BIT_SET_VALUE(out, xfer_ll_params.ce_vio, VPRCSR_NORDIC_OUT_LOW);
 		}
-	}
-
-	/* Counter settings */
-	nrf_vpr_csr_vtim_count_mode_set(0, NRF_VPR_CSR_VTIM_COUNT_RELOAD);
-	nrf_vpr_csr_vtim_simple_counter_top_set(0, xfer_ll_params.counter_top);
-
-	/* Set number of shifts before OUTB needs to be updated.
-	 * First shift needs to be increased by 1.
-	 */
-	nrf_vpr_csr_vio_shift_cnt_out_set(xfer_ll_params.word_size / 4);
-	nrf_vpr_csr_vio_shift_cnt_out_buffered_set(xfer_ll_params.word_size / 4 - 1);
-
-	/* Enable CS */
-	out = nrf_vpr_csr_vio_out_get();
-	out &= ~PIN_OUT_HIGH_MASK(VIO(NRFE_MSPI_CS0_PIN_NUMBER));
-	out |= xfer_ll_params.ce_enable_state ? PIN_OUT_HIGH_MASK(VIO(NRFE_MSPI_CS0_PIN_NUMBER))
-					      : PIN_OUT_LOW_MASK(VIO(NRFE_MSPI_CS0_PIN_NUMBER));
-	nrf_vpr_csr_vio_out_set(out);
-
-	/* Start counter */
-	nrf_vpr_csr_vtim_simple_counter_set(0, 3 * xfer_ll_params.counter_top);
-
-	/* Send data */
-	for (uint8_t i = 0; i < xfer_ll_params.data_len; i++) {
-		nrf_vpr_csr_vio_out_buffered_reversed_byte_set(xfer_ll_params.data_to_send[i]);
-	}
-
-	/* Clear all bits, wait until the last word is sent */
-	nrf_vpr_csr_vio_out_buffered_set(0);
-
-	/* Final configuration */
-	out_mode.mode = NRF_VPR_CSR_VIO_SHIFT_NONE;
-	nrf_vpr_csr_vio_mode_out_buffered_set(&out_mode);
-	nrf_vpr_csr_vio_mode_in_buffered_set(NRF_VPR_CSR_VIO_MODE_IN_CONTINUOUS);
-
-	/* Disable CS */
-	if (!xfer_ll_params.ce_hold) {
-		out = nrf_vpr_csr_vio_out_get();
-		out &= ~(PIN_OUT_HIGH_MASK(VIO(NRFE_MSPI_CS0_PIN_NUMBER)) |
-			 PIN_OUT_HIGH_MASK(VIO(NRFE_MSPI_SCK_PIN_NUMBER)));
-		out |= xfer_ll_params.ce_enable_state
-			       ? PIN_OUT_LOW_MASK(VIO(NRFE_MSPI_CS0_PIN_NUMBER))
-			       : PIN_OUT_HIGH_MASK(VIO(NRFE_MSPI_CS0_PIN_NUMBER));
 		nrf_vpr_csr_vio_out_set(out);
 	}
 
 	/* Stop counter */
 	nrf_vpr_csr_vtim_count_mode_set(0, NRF_VPR_CSR_VTIM_COUNT_STOP);
-}
+}
```

Comment:
No new line at the end of file. Also in other files.

Reply:
Still in [applications/sdp/mspi/boards/nrf54l15dk_nrf54l15_cpuflpr.conf](https://github.com/nrfconnect/sdk-nrf/pull/19449/commits/8b75e8be708183d70dd050490a45439713278e12#diff-67240772fe966e4a791ec269fc4de4ff91f9449febc2bbbce88ff9da735018a5)